### PR TITLE
Add graal info to module file

### PR DESCRIPTION
**Pull request by tobiasdiez**
As mentioned by @michel-kraemer in https://github.com/JabRef/jabref#issuecomment-590334280 the error might result from a non-present module description. Not sure if the one I took covers everything. (@michel-kraemer do you remember what to add exactly?)

- [ ] Change in CHANGELOG.md described (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for bigger UI changes)
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not: Issue created at <https://github.com/JabRef/user-documentation/issues>.

### CODE REVIEW

**Comment by Siedlerchr** [comment]
/home/runner/work/jabref/jabref/src/main/java/module-info.java:67: error: module not found: org.graalvm.js.scriptengine

    requires transitive org.graalvm.js.scriptengine;

**Comment by Siedlerchr** [comment]
Still no sucess:

Btw, you can directly test this locally if you just call gradlew jlink and then run it from the bat
```
java.lang.Exception: java.lang.InternalError: java.lang.IllegalAccessException: class com.oracle.truffle.api.impl.DefaultTruffleRuntime$Loader (in module org.graalvm.truffle) cannot access class jdk.vm.ci.services.Services (in module jdk.internal.vm.ci) because module jdk.internal.vm.ci does not export jdk.vm.ci.services to module org.graalvm.truffle
        at org.jabref.gui.util.DefaultTaskExecutor.convertToException(Unknown Source) ~[org.jabref:?]
        at org.jabref.gui.util.DefaultTaskExecutor.lambda$getJavaFXTask$4(Unknown Source) ~[org.jabref:?]
        at com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(Unknown Source) ~[org.jabref.merged.module:?]
        at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source) ~[org.jabref.merged.module:?]
        at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source) ~[org.jabref.merged.module:?]
        at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source) ~[org.jabref.merged.module:?]
        at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source) ~[org.jabref.merged.module:?]
        at com.sun.javafx.event.EventUtil.fireEventImpl(Unknown Source) ~[org.jabref.merged.module:?]
        at com.sun.javafx.event.EventUtil.fireEvent(Unknown Source) ~[org.jabref.merged.module:?]
        at javafx.event.Event.fireEvent(Unknown Source) ~[org.jabref.merged.module:?]
        at javafx.concurrent.EventHelper.fireEvent(Unknown Source) ~[org.jabref.merged.module:?]
        at javafx.concurrent.Task.fireEvent(Unknown Source) ~[org.jabref.merged.module:?]
        at javafx.concurrent.Task.setState(Unknown Source) ~[org.jabref.merged.module:?]
        at javafx.concurrent.Task$TaskCallable.lambda$call$2(Unknown Source) ~[org.jabref.merged.module:?]
        at com.sun.javafx.application.PlatformImpl.lambda$runLater$10(Unknown Source) ~[org.jabref.merged.module:?]
        at java.security.AccessController.doPrivileged(Unknown Source) ~[?:?]
        at com.sun.javafx.application.PlatformImpl.lambda$runLater$11(Unknown Source) ~[org.jabref.merged.module:?]
        at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(Unknown Source) ~[org.jabref.merged.module:?]
        at com.sun.glass.ui.win.WinApplication._runLoop(Native Method) ~[org.jabref.merged.module:?]
        at com.sun.glass.ui.win.WinApplication.lambda$runLoop$3(Unknown Source) ~[org.jabref.merged.module:?]
        at java.lang.Thread.run(Unknown Source) [?:?]
Caused by: java.lang.InternalError: java.lang.IllegalAccessException: class com.oracle.truffle.api.impl.DefaultTruffleRuntime$Loader (in module org.graalvm.truffle) cannot access class jdk.vm.ci.services.Services (in module jdk.internal.vm.ci) because module jdk.internal.vm.ci does not export jdk.vm.ci.services to module org.graalvm.truffle
        at com.oracle.truffle.api.impl.DefaultTruffleRuntime$Loader.load(Unknown Source) ~[org.graalvm.truffle:?]
        at com.oracle.truffle.api.impl.DefaultTruffleRuntime.getCapability(Unknown Source) ~[org.graalvm.truffle:?]
        at com.oracle.truffle.api.impl.TruffleLocator.loaders(Unknown Source) ~[org.graalvm.truffle:?]
        at com.oracle.truffle.api.impl.Accessor.loaders(Unknown Source) ~[org.graalvm.truffle:?]
        at com.oracle.truffle.polyglot.EngineAccessor.allLoaders(Unknown Source) ~[org.graalvm.truffle:?]
        at com.oracle.truffle.polyglot.LanguageCache.createLanguages(Unknown Source) ~[org.graalvm.truffle:?]
        at com.oracle.truffle.polyglot.LanguageCache.languages(Unknown Source) ~[org.graalvm.truffle:?]
        at com.oracle.truffle.polyglot.PolyglotEngineImpl.languages(Unknown Source) ~[org.graalvm.truffle:?]
        at com.oracle.truffle.polyglot.PolyglotEngineImpl.initializeLanguages(Unknown Source) ~[org.graalvm.truffle:?]
        at com.oracle.truffle.polyglot.PolyglotEngineImpl.<init>(Unknown Source) ~[org.graalvm.truffle:?]
        at com.oracle.truffle.polyglot.PolyglotEngineImpl.<init>(Unknown Source) ~[org.graalvm.truffle:?]
        at com.oracle.truffle.polyglot.PolyglotImpl.buildEngine(Unknown Source) ~[org.graalvm.truffle:?]
        at org.graalvm.polyglot.Engine$Builder.build(Unknown Source) ~[org.graalvm.sdk:?]
        at org.graalvm.polyglot.Context$Builder.build(Unknown Source) ~[org.graalvm.sdk:?]
        at de.undercouch.citeproc.script.GraalScriptRunner.<init>(Unknown Source) ~[org.jabref.merged.module:?]
        at de.undercouch.citeproc.script.ScriptRunnerFactory.createGraalRunner(Unknown Source) ~[org.jabref.merged.module:?]
        at de.undercouch.citeproc.script.ScriptRunnerFactory.createRunner(Unknown Source) ~[org.jabref.merged.module:?]
        at de.undercouch.citeproc.CSL.getRunner(Unknown Source) ~[org.jabref.merged.module:?]
        at de.undercouch.citeproc.CSL.<init>(Unknown Source) ~[org.jabref.merged.module:?]
        at de.undercouch.citeproc.CSL.<init>(Unknown Source) ~[org.jabref.merged.module:?]
        at de.undercouch.citeproc.CSL.<init>(Unknown Source) ~[org.jabref.merged.module:?]
        at org.jabref.logic.citationstyle.CSLAdapter.initialize(Unknown Source) ~[org.jabref:?]
        at org.jabref.logic.citationstyle.CSLAdapter.makeBibliography(Unknown Source) ~[org.jabref:?]
        at org.jabref.logic.citationstyle.CitationStyleGenerator.generateCitations(Unknown Source) ~[org.jabref:?]
        at org.jabref.logic.citationstyle.CitationStyleGenerator.generateCitation(Unknown Source) ~[org.jabref:?]
        at org.jabref.logic.citationstyle.CitationStylePreviewLayout.generatePreview(Unknown Source) ~[org.jabref:?]
        at org.jabref.gui.preview.PreviewViewer.lambda$update$3(Unknown Source) ~[org.jabref:?]
        at org.jabref.gui.util.BackgroundTask$1.call(Unknown Source) ~[org.jabref:?]
        at org.jabref.gui.util.DefaultTaskExecutor$1.call(Unknown Source) ~[org.jabref:?]
        at javafx.concurrent.Task$TaskCallable.call(Unknown Source) ~[org.jabref.merged.module:?]
        at java.util.concurrent.FutureTask.run(Unknown Source) ~[?:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) ~[?:?]
        at java.util.concurrent.FutureTask.run(Unknown Source) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) ~[?:?]
        ... 1 more
Caused by: java.lang.IllegalAccessException: class com.oracle.truffle.api.impl.DefaultTruffleRuntime$Loader (in module org.graalvm.truffle) cannot access class jdk.vm.ci.services.Services (in module jdk.internal.vm.ci) because module jdk.internal.vm.ci does not export jdk.vm.ci.services to module org.graalvm.truffle
        at jdk.internal.reflect.Reflection.newIllegalAccessException(Unknown Source) ~[?:?]
        at java.lang.reflect.AccessibleObject.checkAccess(Unknown Source) ~[?:?]
        at java.lang.reflect.Method.invoke(Unknown Source) ~[?:?]
        at com.oracle.truffle.api.impl.DefaultTruffleRuntime$Loader.load(Unknown Source) ~[org.graalvm.truffle:?]
        at com.oracle.truffle.api.impl.DefaultTruffleRuntime.getCapability(Unknown Source) ~[org.graalvm.truffle:?]
        at com.oracle.truffle.api.impl.TruffleLocator.loaders(Unknown Source) ~[org.graalvm.truffle:?]
        at com.oracle.truffle.api.impl.Accessor.loaders(Unknown Source) ~[org.graalvm.truffle:?]
        at com.oracle.truffle.polyglot.EngineAccessor.allLoaders(Unknown Source) ~[org.graalvm.truffle:?]
        at com.oracle.truffle.polyglot.LanguageCache.createLanguages(Unknown Source) ~[org.graalvm.truffle:?]
        at com.oracle.truffle.polyglot.LanguageCache.languages(Unknown Source) ~[org.graalvm.truffle:?]
        at com.oracle.truffle.polyglot.PolyglotEngineImpl.languages(Unknown Source) ~[org.graalvm.truffle:?]
        at com.oracle.truffle.polyglot.PolyglotEngineImpl.initializeLanguages(Unknown Source) ~[org.graalvm.truffle:?]
        at com.oracle.truffle.polyglot.PolyglotEngineImpl.<init>(Unknown Source) ~[org.graalvm.truffle:?]
        at com.oracle.truffle.polyglot.PolyglotEngineImpl.<init>(Unknown Source) ~[org.graalvm.truffle:?]
        at com.oracle.truffle.polyglot.PolyglotImpl.buildEngine(Unknown Source) ~[org.graalvm.truffle:?]
        at org.graalvm.polyglot.Engine$Builder.build(Unknown Source) ~[org.graalvm.sdk:?]
        at org.graalvm.polyglot.Context$Builder.build(Unknown Source) ~[org.graalvm.sdk:?]
        at de.undercouch.citeproc.script.GraalScriptRunner.<init>(Unknown Source) ~[org.jabref.merged.module:?]
        at de.undercouch.citeproc.script.ScriptRunnerFactory.createGraalRunner(Unknown Source) ~[org.jabref.merged.module:?]
        at de.undercouch.citeproc.script.ScriptRunnerFactory.createRunner(Unknown Source) ~[org.jabref.merged.module:?]
        at de.undercouch.citeproc.CSL.getRunner(Unknown Source) ~[org.jabref.merged.module:?]
        at de.undercouch.citeproc.CSL.<init>(Unknown Source) ~[org.jabref.merged.module:?]
        at de.undercouch.citeproc.CSL.<init>(Unknown Source) ~[org.jabref.merged.module:?]
        at de.undercouch.citeproc.CSL.<init>(Unknown Source) ~[org.jabref.merged.module:?]
        at org.jabref.logic.citationstyle.CSLAdapter.initialize(Unknown Source) ~[org.jabref:?]
        at org.jabref.logic.citationstyle.CSLAdapter.makeBibliography(Unknown Source) ~[org.jabref:?]
        at org.jabref.logic.citationstyle.CitationStyleGenerator.generateCitations(Unknown Source) ~[org.jabref:?]
        at org.jabref.logic.citationstyle.CitationStyleGenerator.generateCitation(Unknown Source) ~[org.jabref:?]
        at org.jabref.logic.citationstyle.CitationStylePreviewLayout.generatePreview(Unknown Source) ~[org.jabref:?]
        at org.jabref.gui.preview.PreviewViewer.lambda$update$3(Unknown Source) ~[org.jabref:?]
        at org.jabref.gui.util.BackgroundTask$1.call(Unknown Source) ~[org.jabref:?]
        at org.jabref.gui.util.DefaultTaskExecutor$1.call(Unknown Source) ~[org.jabref:?]
        at javafx.concurrent.Task$TaskCallable.call(Unknown Source) ~[org.jabref.merged.module:?]
        at java.util.concurrent.FutureTask.run(Unknown Source) ~[?:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) ~[?:?]
        at java.util.concurrent.FutureTask.run(Unknown Source) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) ~[?:?]
```

**Comment by Siedlerchr** [comment]
Seems to be a bug in graal which got resolved in 19.3.0: https://github.com/oracle/graal/issues/1751
@michel-kraemer  I merged your PR into this branch to test the things with your new snapshot. 
Would it be possible to update to the new graal version?
https://github.com/michel-kraemer/citeproc-java/blob/master/citeproc-java/build.gradle#L26

**Comment by Siedlerchr** [comment]
Yippieh, it works now! 🚨 🎉 😀 

```
21:02:56.890 [JavaFX Application Thread] ERROR org.jabref.gui.preview.PreviewViewer - Error while generating citation style
java.lang.NullPointerException: null
        at de.undercouch.citeproc.csl.internal.rendering.SDate.renderInternal(SDate.java:58) ~[citeproc-java-2.1.0-SNAPSHOT.jar:?]
        at de.undercouch.citeproc.csl.internal.behavior.Affixes.accept(Affixes.java:38) ~[citeproc-java-2.1.0-SNAPSHOT.jar:?]
        at de.undercouch.citeproc.csl.internal.behavior.Affixes.accept(Affixes.java:14) ~[citeproc-java-2.1.0-SNAPSHOT.jar:?]
        at de.undercouch.citeproc.csl.internal.behavior.Behavior.lambda$wrap$0(Behavior.java:21) ~[citeproc-java-2.1.0-SNAPSHOT.jar:?]
        at de.undercouch.citeproc.csl.internal.rendering.SDate.render(SDate.java:49) ~[citeproc-java-2.1.0-SNAPSHOT.jar:?]
        at de.undercouch.citeproc.csl.internal.SRenderingElementContainer.lambda$render$0(SRenderingElementContainer.java:57) ~[citeproc-java-2.1.0-SNAPSHOT.jar:?]
        at java.util.ArrayList.forEach(ArrayList.java:1507) ~[?:?]
        at de.undercouch.citeproc.csl.internal.SRenderingElementContainer.render(SRenderingElementContainer.java:57) ~[citeproc-java-2.1.0-SNAPSHOT.jar:?]
        at de.undercouch.citeproc.csl.internal.rendering.SText.renderInternal(SText.java:102) ~[citeproc-java-2.1.0-SNAPSHOT.jar:?]
        at de.undercouch.citeproc.csl.internal.behavior.TextCase.accept(TextCase.java:31) ~[citeproc-java-2.1.0-SNAPSHOT.jar:?]
        at de.undercouch.citeproc.csl.internal.behavior.TextCase.accept(TextCase.java:17) ~[citeproc-java-2.1.0-SNAPSHOT.jar:?]
        at de.undercouch.citeproc.csl.internal.behavior.Behavior.lambda$wrap$0(Behavior.java:21) ~[citeproc-java-2.1.0-SNAPSHOT.jar:?]
        at de.undercouch.citeproc.csl.internal.behavior.Quotes.accept(Quotes.java:41) ~[citeproc-java-2.1.0-SNAPSHOT.jar:?]
        at de.undercouch.citeproc.csl.internal.behavior.Quotes.accept(Quotes.java:14) ~[citeproc-java-2.1.0-SNAPSHOT.jar:?]
        at de.undercouch.citeproc.csl.internal.behavior.Behavior.lambda$wrap$0(Behavior.java:21) ~[citeproc-java-2.1.0-SNAPSHOT.jar:?]
        at de.undercouch.citeproc.csl.internal.behavior.Affixes.accept(Affixes.java:38) ~[citeproc-java-2.1.0-SNAPSHOT.jar:?]
        at de.undercouch.citeproc.csl.internal.behavior.Affixes.accept(Affixes.java:14) ~[citeproc-java-2.1.0-SNAPSHOT.jar:?]
        at de.undercouch.citeproc.csl.internal.behavior.Behavior.lambda$wrap$0(Behavior.java:21) ~[citeproc-java-2.1.0-SNAPSHOT.jar:?]
        at de.undercouch.citeproc.csl.internal.rendering.SText.render(SText.java:57) ~[citeproc-java-2.1.0-SNAPSHOT.jar:?]
        at de.undercouch.citeproc.csl.internal.SLayout.renderInternal(SLayout.java:43) ~[citeproc-java-2.1.0-SNAPSHOT.jar:?]
        at de.undercouch.citeproc.csl.internal.behavior.Affixes.accept(Affixes.java:38) ~[citeproc-java-2.1.0-SNAPSHOT.jar:?]
        at de.undercouch.citeproc.csl.internal.behavior.Affixes.accept(Affixes.java:14) ~[citeproc-java-2.1.0-SNAPSHOT.jar:?]
        at de.undercouch.citeproc.csl.internal.behavior.Behavior.lambda$wrap$0(Behavior.java:21) ~[citeproc-java-2.1.0-SNAPSHOT.jar:?]
        at de.undercouch.citeproc.csl.internal.SLayout.render(SLayout.java:25) ~[citeproc-java-2.1.0-SNAPSHOT.jar:?]
        at de.undercouch.citeproc.csl.internal.SBibliography.render(SBibliography.java:66) ~[citeproc-java-2.1.0-SNAPSHOT.jar:?]
        at de.undercouch.citeproc.CSL.makeBibliography(CSL.java:1331) ~[citeproc-java-2.1.0-SNAPSHOT.jar:?]
        at de.undercouch.citeproc.CSL.makeBibliography(CSL.java:1218) ~[citeproc-java-2.1.0-SNAPSHOT.jar:?]
        at org.jabref.logic.citationstyle.CSLAdapter.makeBibliography(CSLAdapter.java:56) ~[main/:?]
        at org.jabref.logic.citationstyle.CitationStyleGenerator.generateCitations(CitationStyleGenerator.java:56) ~[main/:?]
        at org.jabref.logic.citationstyle.CitationStyleGenerator.generateCitation(CitationStyleGenerator.java:47) ~[main/:?]
        at org.jabref.logic.citationstyle.CitationStylePreviewLayout.generatePreview(CitationStylePreviewLayout.java:15) ~[main/:?]
        at org.jabref.gui.preview.PreviewViewer.lambda$update$3(PreviewViewer.ja
```

For this entry:

```
@Article{bbbbbbbb,
  author          = {Trapnell, Cole and Pachter, Lior and Salzberg, Steven L.},
  journal         = {2D Materials},
  title           = {{TopHat}: discovering splice junctions with {RNA}-Seq},
  year            = {2009},
  issn            = {1866-0452},
  month           = {mar},
  number          = {9},
  pages           = {1105--1111},
  volume          = {25},
  citation-subset = {IM},
  country         = {Germany},
  created         = {2016-05-06},
  date            = {2016},
  doi             = {10.1093/bbbb/btp120},
  issn-linking    = {1866-0452},
  issue           = {15},
  keywords        = {rank4},
  nlm             = {PMC4860871},
  nlm-id          = {101475967},
  pii             = {arztebl.2016.0261},
  pmc             = {PMC4860871},
  pmid            = {27151464},
  publisher       = {Oxford University Press ({OUP})},
  pubmodel        = {Print},
  pubstatus       = {ppublish},
  revised         = {2016-05-23},
  timestamp       = {2017-07-15},
  url             = {http://www.google.de},
}
```

**Comment by Siedlerchr** [comment]
All tests performed with `./gradlew jlink`

File: build.gradle [hunk 1]
```diff
@@ -181,7 +181,9 @@ dependencies {
     implementation group: 'org.apache.logging.log4j', name: 'log4j-core', version: '3.0.0-SNAPSHOT'
     annotationProcessor group: 'org.apache.logging.log4j', name: 'log4j-core', version: '3.0.0-SNAPSHOT'
 
-    implementation 'de.undercouch:citeproc-java:2.0.0'
+
```
Comment (koppor) [inline build.gradle]: Please no two empty lines separation

File: build.gradle [hunk 2]
```diff
@@ -546,7 +548,7 @@ task deleteInstallerTemp(type: Delete) {
 
 jpackage.dependsOn deleteInstallerTemp
 jlink {
-    options = ['--strip-debug', '--compress', '2', '--no-header-files', '--no-man-pages']
+    options = ['--compress', '2', '--no-header-files', '--no-man-pages']
```
Comment (koppor) [inline build.gradle]: oh, wow, we still need `debug`? (Because `--strip-debug` was removed`)

**Comment by michel-kraemer** [comment]
@Siedlerchr

> @michel-kraemer I merged your PR into this branch to test the things with your new snapshot.
> Would it be possible to update to the new graal version?

Sure. Shouldn't be a larger problem. Do you need it urgently? If so, I can do it on the weekend. If not, I'd prefer to do it on the next weekend after my paper deadline :)

**Comment by Siedlerchr** [comment]
@michel-kraemer  As it's working now in general, I don't think it's anylonger necessary to update graal urgently.

**Comment by michel-kraemer** [comment]
@Siedlerchr Great! Let me know if you find bugs in the pure Java implementation. Also, if you need a fixed version (i.e. a beta.1 instead of a SNAPSHOT), just contact me.

**Comment by Siedlerchr** [comment]
@michel-kraemer  This entry created an exception https://github.com/JabRef/jabref/pull/6010#issuecomment-591618466

And when creating a new entry with empty fields I got an exception as well. I will check later for details again. But also tested with another database and that worked fine so far.

**Comment by michel-kraemer** [comment]
@Siedlerchr May I ask you to create issues in the citeproc-java repo (https://github.com/michel-kraemer/citeproc-java), so we can keep track of the fixes? Thanks!

File: src/main/java/module-info.java [hunk 3]
```diff
@@ -61,7 +61,11 @@
     requires jbibtex;
     requires citeproc.java;
     requires antlr.runtime;
-    requires org.graalvm.js;
+    requires org.graalvm.truffle;
+    requires org.graalvm.sdk;
+    requires transitive org.graalvm.js;
+    requires java.scripting;
```
Comment (koppor) [inline src/main/java/module-info.java:67]: I thought that `2.1.0-SNAPSHOT` does not require JavaScript anymore (see also https://github.com/michel-kraemer/citeproc-java/issues/92). @tobiasdiez Do you remember why you had to add these dependencies?

**Pull request merged**